Enable styleCheck:usages

Menduist opened the pull request.

cheatfate commented on nimcrypto/blowfish.nim:
```diff
@@ -461,24 +461,24 @@ proc init*(ctx: var BlowfishContext, key: ptr byte, nkey: int) {.inline.} =
   blowfishDecrypt(ctx, input, output)
 
-proc encrypt*(ctx: var BlowfishContext, inbytes: ptr byte,
+proc encrypt*(ctx: var BlowfishContext, inBytes: ptr byte,
```
This rename is not going to be accepted, leave it as it was before, please.

cheatfate commented on tests/testripemd.nim:
```diff
@@ -262,41 +262,41 @@ suite "RipeMD Tests":
-      check1286 == Ripemd128C[5]
-      check1287 == Ripemd128C[6]
-      check1288 == Ripemd128C[7]
+      check1281 == RipeMD128C[0]
```
Please do not touch this constants.

cheatfate commented on tests/testscrypt.nim:
```diff
@@ -153,7 +153,7 @@ suite "Scrypt KDF tests suite":
     let stringarg = scrypt(password="password", salt="NaCl", N=1024, r=8, p=16, keyLen=32)
-    let openarrayarg = scrypt(fromHex("70617373776F7264"), fromHex("4E61436C"), N=1024, r=8, p=16, keyLen=32)
-    check stringarg == openarrayarg
+    let openArrayarg = scrypt(fromHex("70617373776F7264"), fromHex("4E61436C"), N=1024, r=8, p=16, keyLen=32)
```
do not touch this variable name.

cheatfate commented on tests/testsha2.nim:
```diff
@@ -102,8 +102,8 @@ suite "SHA2 Tests":
       ctx512.sizeBlock == 128
-      ctx512224.sizeBlock == 128
-      ctx512256.sizeBlock == 128
+      ctx512_224.sizeBlock == 128
```
ditto

cheatfate commented on tests/testsha2.nim:
```diff
@@ -117,8 +117,8 @@ suite "SHA2 Tests":
       ctx512.sizeDigest == 64
-      ctx512224.sizeDigest == 28
-      ctx512256.sizeDigest == 32
+      ctx512_224.sizeDigest == 28
```
ditto

cheatfate commented on tests/testripemd.nim:
```diff
@@ -17,7 +17,7 @@ suite "RipeMD Tests":
   ]
 
-  const Ripemd128C = [
+  const RipeMD128C = [
```
Do not rename constants, please.

Menduist replied:
This is used as `RipeMD128C` on line 129, 133, 160, etc
https://github.com/cheatfate/nimcrypto/blob/a5742a9a214ac33f91615f3862c7b099aec43b00/tests/testripemd.nim#L129

Since the other two variables are in the format RipeMDXXXC, I renamed this one to be consistent with the other two, instead of changing the usage

cheatfate commented:
I would recommend to stick with `openarray` to `openArray` rename only, all other renames (arguments, variables, constants) are not going to be accepted.

Menduist commented on tests/testsha2.nim:
```diff
@@ -86,8 +86,8 @@ suite "SHA2 Tests":
   var ctx512: sha512
-  var ctx512224: sha512_224
-  var ctx512256: sha512_256
+  var ctx512_224: sha512_224
```
https://github.com/cheatfate/nimcrypto/blob/a5742a9a214ac33f91615f3862c7b099aec43b00/tests/testsha2.nim#L357

It's used here as `ctx512_224`, and it seemed more consistent to put the underscore everywhere, but we can switch to `ctx512224` everywhere if you want

Menduist commented:
I renamed arguments, variables & constants because they were used inconsistently and caused compilation error. I updated the code to remove some useless changes I did, let me know if it's ok

Menduist commented on nimcrypto/sysrand.nim:
```diff
@@ -109,7 +109,7 @@ elif defined(linux):
   var gSystemRng {.threadvar.}: SystemRng ## System thread global RNG
 
-  proc newSystemRNG(): SystemRng =
+  proc newSystemRng(): SystemRng =
```
Used as newSystemRng here:
https://github.com/cheatfate/nimcrypto/blob/a5742a9a214ac33f91615f3862c7b099aec43b00/nimcrypto/sysrand.nim#L125

cheatfate commented:
Public and private procedure names as well as variable name must not be changed.

Menduist commented:
Then it won't compile, styleCheck applies to everything, not just public stuff

Menduist commented:
Seems like #55 was enough
